### PR TITLE
PYMT-1087 search fill

### PR DESCRIPTION
**Pull request by merk**
The goal of this PR is to refactor out the "filling" of an index into a thing that can be customised on a per handler basis.

### CODE REVIEW

File: src/Interfaces/PopulatorInterface.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+     * @return void
+     */
+    public function populate(
+        TransformableSearchHandlerInterface $handler,
+        string $indexSuffix,
+        int $batchSize
+    ): void;
+
+    /**
+     * Populates a handler's index with an array of objects.
```
Comment (ketanp77) [inline src/Interfaces/PopulatorInterface.php:24]: The method description is identical to `populate()`, could it be different for `populateWith()`?

File: tests/ClientTest.php [hunk 2]
```diff
@@ -99,7 +100,7 @@ public function testBulkCallableResolution(): void
         $stub = new CallableResponseClientStub();
         $client = $this->createInstance($stub);
 
-        $client->bulkUpdate('index', ['1' => 'document']);
+        $client->bulkUpdate([new DocumentUpdate('index', '1', 'document')]);
 
         // If call was successful there should be no return/exception
         $this->addToAssertionCount(1);
```
Comment (ketanp77) [inline tests/ClientTest.php:106]: We have further assertion, so I guess we can remove `addToAssertionCount`?

**Review by ketanp77 — DISMISSED** [review]
Looks 👍 , only few comments

File: src/Interfaces/PopulatorInterface.php (hunk 1 above)
Comment (merk) [reply src/Interfaces/PopulatorInterface.php:24]: Good catch

File: tests/ClientTest.php (hunk 2 above)
Comment (merk) [reply tests/ClientTest.php:106]: This one looks more complicated because it involves dealing with futures

**Pull request merged**